source/extensions/quic_listeners/quiche/platform/quic_logging_impl.h

```diff
@@ -56,13 +56,16 @@
 #define QUIC_LOG_WARNING_IS_ON_IMPL() quic::IsLogLevelEnabled(quic::WARNING)
 #define QUIC_LOG_ERROR_IS_ON_IMPL() quic::IsLogLevelEnabled(quic::ERROR)
 
+// CHECK and DCHECK are only used by quiche code, use ASSERT instead in envoy code.
```

Comment:
Maybe identify in particular which BUILD target depends on this?

Reply:
Can you add the pointer to the doc somewhere obvious, maybe here too, as this is very counterintuitive why this is needed. I confirmed though that commenting this out breaks the build in weird ways.

Someone thing in the code has got to say why there is an Quiche -> Envoy -> Quiche dependency chain as this is not expected.